APMrover2/Steering.cpp

```diff
@@ -231,68 +234,61 @@ void Rover::calc_nav_steer() {
+        SRV_Channels::set_output_pwm(SRV_Channel::k_steering,channel_steer->read());
+        SRV_Channels::set_output_pwm(SRV_Channel::k_throttle,channel_throttle->read());
         if (failsafe.bits & FAILSAFE_EVENT_THROTTLE) {
             // suppress throttle if in failsafe and manual
-            channel_throttle->set_radio_out(channel_throttle->get_radio_trim());
+            SRV_Channels::set_output_pwm(SRV_Channel::k_throttle,channel_throttle->get_radio_trim());
             // suppress steer if in failsafe and manual and skid steer mode
             if (g.skid_steer_out) {
-                channel_steer->set_radio_out(channel_steer->get_radio_trim());
+                SRV_Channels::set_output_pwm(SRV_Channel::k_steering,channel_steer->get_radio_trim());
             }
         }
     } else {
-        channel_steer->calc_pwm();
         if (in_reverse) {
-            channel_throttle->set_servo_out(constrain_int16(channel_throttle->get_servo_out(),
+            SRV_Channels::set_output_scaled(SRV_Channel::k_throttle,constrain_int16(SRV_Channels::get_output_scaled(SRV_Channel::k_throttle),
                                           -g.throttle_max,
                                           -g.throttle_min));
         } else {
-            channel_throttle->set_servo_out(constrain_int16(channel_throttle->get_servo_out(),
+            SRV_Channels::set_output_scaled(SRV_Channel::k_throttle,constrain_int16(SRV_Channels::get_output_scaled(SRV_Channel::k_throttle),
                                           g.throttle_min.get(),
                                           g.throttle_max.get()));
         }
 
         if ((failsafe.bits & FAILSAFE_EVENT_THROTTLE) && control_mode < AUTO) {
             // suppress throttle if in failsafe
-            channel_throttle->set_servo_out(0);
+            SRV_Channels::set_output_scaled(SRV_Channel::k_throttle,0);
             // suppress steer if in failsafe and skid steer mode
             if (g.skid_steer_out) {
-                channel_steer->set_servo_out(0);
+                SRV_Channels::set_output_scaled(SRV_Channel::k_steering,0);
             }
         }
 
         if (!hal.util->get_soft_armed()) {
-            channel_throttle->set_servo_out(0);
+            SRV_Channels::set_output_scaled(SRV_Channel::k_throttle,0);
             // suppress steer if in failsafe and skid steer mode
             if (g.skid_steer_out) {
-                channel_steer->set_servo_out(0);
+                SRV_Channels::set_output_scaled(SRV_Channel::k_steering,0);
             }
         }
 
-        // convert 0 to 100% into PWM
-        channel_throttle->calc_pwm();
-
         // limit throttle movement speed
         throttle_slew_limit(last_throttle);
     }
 
     // record last throttle before we apply skid steering
-    last_throttle = channel_throttle->get_radio_out();
+    SRV_Channels::get_output_pwm(SRV_Channel::k_throttle, last_throttle);
```

Comment:
The reason for the new API is there may in fact be no channel marked as k_throttle, so no answer is possible.